spec/compiler/semantic/tuple_spec.cr

```diff
@@ -287,7 +287,18 @@ describe "Semantic: tuples" do
         pos += {0, 0}
       end
       ),
-      "tuple too big"
+      "tuple size cannot be greater than 300 (size is 301)"
+  end
+
+  it "errors on named tuple too big" do
+    assert_error %(
+      { #{String.build do |io|
+            301.times do |i|
```

Comment:
Maybe make this number a bit higher to ensure `size is 301` isn't a hard coded value.